src/Frame.h

```diff
@@ -218,6 +218,29 @@ class Frame_REQUEST_CHANNEL : public Frame_REQUEST_Base {
       : Frame_REQUEST_CHANNEL(streamId, flags, 0, std::move(payload)) {}
 };
 
+class Frame_REQUEST_RESPONSE {
+ public:
+  static constexpr bool Trait_CarriesAllowance = false;
+
+  Frame_REQUEST_RESPONSE() = default;
+  Frame_REQUEST_RESPONSE(StreamId streamId, FrameFlags flags, Payload payload)
+      : header_(
+            FrameType::REQUEST_RESPONSE,
+            flags | payload.getFlags(),
+            streamId),
+        payload_(std::move(payload)) {
+    payload_.checkFlags(header_.flags_); // to verify the client didn't set
```

Comment:
agree. The metadata flag should be sent only by presence of payload.metadata. Just need to make sure the user can't set it from outside.
We don't have to treat this in this PR.